src/v/kafka/client/consumer.cc

```diff
@@ -475,22 +478,60 @@ ss::future<fetch_response> consumer::fetch(
       detail::reduce_fetch_response);
 }
 
+template<typename request_factory, typename response_t>
+ss::future<response_t>
+consumer::maybe_process_response_errors(request_factory req, response_t res) {
+    if constexpr (std::is_same_v<response_t, offset_commit_response>) {
+        // Look at partition level for error
+        return ss::make_ready_future<response_t>(std::move(res));
```

Comment:
Before I write the code, I'm thinking that in responses such as `offset_commit_response` we can retry the whole request if one of the topic-partitions had `not_coordinator`. It's naive but it beats returning a partial response.

Any objections?